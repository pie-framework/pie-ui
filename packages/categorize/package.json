{
  "name": "@pie-ui/categorize",
  "version": "1.5.4",
  "description": "",
  "main": "lib/index.js",
  "module": "src/index.js",
  "publishConfig": {
    "access": "public"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@pie-framework/pie-player-events": "^0.1.0",
    "@pie-lib/categorize": "^0.6.18",
    "@pie-lib/correct-answer-toggle": "^2.3.14",
    "@pie-lib/drag": "^1.1.45",
    "@pie-lib/math-rendering": "^2.3.1",
    "@pie-lib/render-ui": "^4.10.3",
    "classnames": "^2.2.5",
    "debug": "^4.1.1",
    "lodash": "^4.17.15",
    "prop-types": "^15.6.1",
    "react": "^16.8.1",
    "react-dnd": "^9.3.2",
    "react-dom": "^16.8.1"
  },
  "devDependencies": {
    "react": "^16.8.1"
  },
  "gitHead": "25b2169d13da6abaaaa962e42f3f6d6ab400ec8b"
}
