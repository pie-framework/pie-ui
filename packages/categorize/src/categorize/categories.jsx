import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { color } from '@pie-lib/render-ui';

import chunk from 'lodash/chunk';

import GridContent from './grid-content';
import Category, { CategoryType } from './category';

export { CategoryType };

export class Categories extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    categories: PropTypes.arrayOf(PropTypes.shape(CategoryType)),
    model: PropTypes.shape({
      categoriesPerRow: PropTypes.number,
      choicesPerRow: PropTypes.number,
    }),
    disabled: PropTypes.bool,
    onDropChoice: PropTypes.func.isRequired,
    onRemoveChoice: PropTypes.func.isRequired,
  };

  static defaultProps = {
    model: {
      categoriesPerRow: 1,
      choicesPerRow: 1
    }
  };

  render() {
    const {
      classes,
      categories,
      model,
      disabled,
      onDropChoice,
      onRemoveChoice,
    } = this.props;
    const { choicesPerRow, categoriesPerRow } = model;

    const columns = choicesPerRow / categoriesPerRow;

    // split categories into an array of arrays (inner array),
    // where each inner array represents how many categories should be displayed on one row
    const chunkedCategories = chunk(categories, categoriesPerRow);

    return (
      <GridContent
        columns={categoriesPerRow}
        className={classes.categories}
        rows={Math.ceil(categories.length / categoriesPerRow) * 2}
      >
        {
          chunkedCategories.map(cat => {
            let items = [];

            // for each inner array of categories, create a row with category titles
            cat.forEach(c => {
              items.push((
                <Typography className={classes.label} key={`category-label-${c.label}`}>
                  <span dangerouslySetInnerHTML={{ __html: c.label }}/>
                </Typography>));
            });

            // if the last row has less categories than max on a row, fill the spaces with divs
            items = items.concat(Array(categoriesPerRow - cat.length).fill(<div/>));

            // for each inner array of categories, create a row with category containers
            cat.forEach((c, index) => {
              const rows = Math.floor(c.choices.length / columns) + 1;

              items.push(<Category
                grid={{ rows, columns }}
                onDropChoice={h => onDropChoice(c.id, h)}
                onRemoveChoice={onRemoveChoice}
                disabled={disabled}
                className={classes.category}
                key={index}
                {...c}
              />);
            });

            // if the last row has less categories than max on a row, fill the spaces with divs
            items = items.concat(Array(categoriesPerRow - cat.length).fill(<div/>));

            return items;
          })
        }
      </GridContent>
    );
  }
}

const styles = theme => ({
  categories: {
    flex: 1
  },
  label: {
    color: color.text(),
    backgroundColor: color.background(),
    textAlign: 'center',
    paddingTop: theme.spacing.unit
  }
});
export default withStyles(styles)(Categories);
