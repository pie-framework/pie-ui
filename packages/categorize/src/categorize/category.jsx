import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Choice from './choice';
import PlaceHolder from './droppable-placeholder';
import { color } from '@pie-lib/render-ui';

export const CategoryType = {
  id: PropTypes.string.isRequired,
  categoryId: PropTypes.string
};

export class Category extends React.Component {
  static propTypes = {
    ...CategoryType,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    classes: PropTypes.object.isRequired,
    onDropChoice: PropTypes.func,
    onRemoveChoice: PropTypes.func
  };

  static defaultProps = {};

  render() {
    const {
      classes,
      className,
      choices,
      disabled,
      onDropChoice,
      onRemoveChoice,
      grid,
      id,
      correct
    } = this.props;

    const names = classNames(classes.category, className);

    const placeholderNames = classNames(
      classes.placeholder,
      correct === false && classes.incorrect
    );

    return (
      <div className={names}>
        <PlaceHolder
          grid={{
            ...grid,
            rowsRepeatValue: 'minmax(60px, auto)'
          }}
          onDropChoice={onDropChoice}
          disabled={disabled}
          className={placeholderNames}
        >
          {choices.map((c, index) => (
            <Choice
              onRemoveChoice={onRemoveChoice}
              disabled={disabled}
              key={index}
              choiceIndex={index}
              categoryId={id}
              {...c}
            />
          ))}
        </PlaceHolder>
      </div>
    );
  }
}
const styles = () => ({
  incorrect: {
    border: `solid 2px ${color.incorrect()}`
  },
  placeholder: {
    minHeight: '60px',
    flex: '1',
    display: 'grid'
  },
  category: {
    display: 'flex',
    flexDirection: 'column'
  }
});
export default withStyles(styles)(Category);
