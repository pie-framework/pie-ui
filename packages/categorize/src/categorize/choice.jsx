import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { DragSource } from 'react-dnd';
import { uid } from '@pie-lib/drag';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import { color } from '@pie-lib/render-ui';
import debug from 'debug';

const log = debug('@pie-ui:categorize:choice');

export const ChoiceType = {
  content: PropTypes.string.isRequired,
  id: PropTypes.string
};

export class Layout extends React.Component {
  static propTypes = {
    ...ChoiceType,
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    correct: PropTypes.bool
  };
  static defaultProps = {};
  render() {
    const {
      classes,
      className,
      content,
      isDragging,
      disabled,
      correct
    } = this.props;

    const rootNames = classNames(
      correct === true && 'correct',
      correct === false && 'incorrect',
      classes.choice,
      isDragging && classes.dragging,
      disabled && classes.disabled,
      className
    );
    const cardNames = classNames(classes.card);
    return (
      <div className={rootNames}>
        <Card className={cardNames}>
          <CardContent classes={{ root: classes.cardRoot }} dangerouslySetInnerHTML={{ __html: content }} />
        </Card>
      </div>
    );
  }
}

const styles = () => ({
  choice: {
    cursor: 'pointer',
    height: '100%',
    width: '100%',
    '&.correct': {
      border: `solid 2px ${color.correct()}`
    },
    '&.incorrect': {
      border: `solid 2px ${color.incorrect()}`
    }
  },
  cardRoot: {
    color: color.text(),
    backgroundColor: color.background(),
    border: '1px solid'
  },
  disabled: {
    cursor: 'not-allowed',
    opacity: '0.6'
  },
  dragging: {
    cursor: 'move'
  },
  card: {
    color: color.text(),
    backgroundColor: color.background(),
    width: '100%'
  }
});

const Styled = withStyles(styles)(Layout);

export class Choice extends React.Component {
  static propTypes = {
    ...ChoiceType,
    connectDragSource: PropTypes.func.isRequired
  };

  render() {
    const {
      connectDragSource,
      id,
      content,
      disabled,
      isDragging,
      correct
    } = this.props;
    return connectDragSource(
      <div style={{ width: '100%', height: 'max-content' }}>
        <Styled
          id={id}
          content={content}
          disabled={disabled}
          correct={correct}
          isDragging={isDragging}
        />
      </div>
    );
  }
}

export const spec = {
  canDrag: props => !props.disabled,
  beginDrag: props => {
    const out = {
      id: props.id,
      categoryId: props.categoryId,
      choiceIndex: props.choiceIndex
    };
    log('[beginDrag] out:', out);
    return out;
  },
  endDrag: (props, monitor) => {
    if (!monitor.didDrop()) {
      const item = monitor.getItem();
      if (item.categoryId) {
        log('wasnt droppped - what to do?');
        props.onRemoveChoice(item);
      }
    }
  }
};

const DraggableChoice = DragSource(
  ({ uid }) => uid,
  spec,
  (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  })
)(Choice);

export default uid.withUid(DraggableChoice);
