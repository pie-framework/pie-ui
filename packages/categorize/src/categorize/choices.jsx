import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Choice, { ChoiceType } from './choice';
export { ChoiceType };
import GridContent from './grid-content';

const Blank = () => <div />;

export class Choices extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    choices: PropTypes.arrayOf(
      PropTypes.oneOfType([
        PropTypes.shape(ChoiceType),
        PropTypes.shape({ empty: PropTypes.bool }),
      ])
    ),
    model: PropTypes.shape({
      choicesPerRow: PropTypes.number,
      choicesLabel: PropTypes.string,
    }),
    disabled: PropTypes.bool,
    choicePosition: PropTypes.string,
  };

  static defaultProps = {
    model: {
      choicesPerRow: 4,
      choicesLabel: '',
    },
  };

  render() {
    const {
      classes,
      choices = [],
      model,
      disabled,
      choicePosition,
    } = this.props;
    let style = {
      textAlign: 'center',
    };

    if (choicePosition === 'left') {
      style.direction = 'rtl';
    }

    return (
      <div className={classes.wrapper}>
        {model.choicesLabel && model.choicesLabel !== '' && (
          <div className={classes.labelHolder}>{model.choicesLabel}</div>
        )}
        <GridContent
          columns={model.choicesPerRow}
          className={classes.choices}
          extraStyle={style}
        >
          {choices.map((c, index) => {
            return c.empty ? (
              <Blank key={index} />
            ) : (
              <Choice
                disabled={disabled}
                className={classes.choice}
                key={index}
                {...c}
              />
            );
          })}
        </GridContent>
      </div>
    );
  }
}

const styles = (theme) => ({
  wrapper: {
    flex: 1,
    padding: theme.spacing.unit,
  },
  choices: {
    paddingTop: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
  },
  labelHolder: {
    margin: '0 auto',
    textAlign: 'center',
    paddingTop: theme.spacing.unit,
  },
});

export default withStyles(styles)(Choices);
