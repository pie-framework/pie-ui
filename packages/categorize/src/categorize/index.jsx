import React from 'react';
import PropTypes from 'prop-types';
import Choices from './choices';
import Categories from './categories';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { withStyles } from '@material-ui/core/styles';
import {
  buildState,
  removeChoiceFromCategory,
  moveChoiceToCategory
} from '@pie-lib/categorize';
import { withDragContext, uid } from '@pie-lib/drag';
import { Feedback, Collapsible } from '@pie-lib/render-ui';

import debug from 'debug';

const log = debug('@pie-ui:categorize');

const removeHTMLTags = html => {
  const tmp = document.createElement('DIV');

  tmp.innerHTML = html;

  const value = tmp.textContent || tmp.innerText || '';

  return value.trim();
};

export class Categorize extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    model: PropTypes.object,
    session: PropTypes.shape({
      answers: PropTypes.arrayOf(
        PropTypes.shape({
          choice: PropTypes.string,
          category: PropTypes.string
        })
      )
    }),
    onAnswersChange: PropTypes.func.isRequired,
    onShowCorrectToggle: PropTypes.func.isRequired
  };

  static defaultProps = {
    disabled: false
  };

  constructor(props) {
    super(props);

    this.state = {
      showCorrect: false
    };
  }

  dropChoice = (categoryId, draggedChoice) => {
    const { session, onAnswersChange } = this.props;
    log(
      '[dropChoice] category: ',
      draggedChoice.categoryId,
      'choice: ',
      draggedChoice
    );

    const answers = moveChoiceToCategory(
      draggedChoice.id,
      draggedChoice.categoryId,
      categoryId,
      draggedChoice.choiceIndex,
      session.answers
    );

    onAnswersChange(answers);
  };

  removeChoice = c => {
    log('[removeChoice]: ', c);
    const { onAnswersChange, session } = this.props;
    const answers = removeChoiceFromCategory(
      c.id,
      c.categoryId,
      c.choiceIndex,
      session.answers
    );
    onAnswersChange(answers);
  };

  UNSAFE_componentWillReceiveProps() {
    this.setState({ showCorrect: false });
  }

  toggleShowCorrect = () =>
    this.setState({ showCorrect: !this.state.showCorrect }, () => {
      this.props.onShowCorrectToggle();
    });

  getPositionDirection = (choicePosition) => {
    let flexDirection;

    switch (choicePosition) {
      case 'left':
        flexDirection = 'row-reverse';
        break;
      case 'right':
        flexDirection = 'row';
        break;
      case 'below':
        flexDirection = 'column';
        break;
      default:
        // above
        flexDirection = 'column-reverse';
        break;
    }

    return flexDirection;
  };

  render() {
    const { classes, model, session } = this.props;
    const { showCorrect } = this.state;
    const { choicesPosition, choicesPerRow, categoriesPerRow } = model;

    const choicePosition = choicesPosition || 'above';

    const style = {
      flexDirection: this.getPositionDirection(choicePosition)
    };

    const { categories, choices, correct } = buildState(
      model.categories,
      model.choices,
      showCorrect ? model.correctResponse : session.answers,
      model.correctResponse
    );

    log('[render] disabled: ', model.disabled);

    const columns = choicesPerRow / categoriesPerRow;

    const maxLength = categories.reduce((acc, c) => {
      if (c.choices.length > acc) {
        return c.choices.length;
      } else {
        return acc;
      }
    }, 0);
    const rows = Math.floor(maxLength / columns) + 1;
    const grid = { rows, columns };
    const { rowLabels } = model;

    return (
      <div>
        {
          model.teacherInstructions && (
            <Collapsible
              labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
              className={classes.collapsible}
            >
              <div dangerouslySetInnerHTML={{ __html: model.teacherInstructions }}/>
            </Collapsible>
          )
        }
        <br />
        <CorrectAnswerToggle
          show={showCorrect || correct === false}
          toggled={showCorrect}
          onToggle={this.toggleShowCorrect}
        />
        {
          removeHTMLTags(model.prompt) &&
          <div
            className={classes.prompt}
            dangerouslySetInnerHTML={{ __html: model.prompt }}
          />
        }
        <div className={classes.categorize} style={style}>
          <div
            style={{
              display: 'flex'
            }}
          >
            {
              rowLabels && (
                <div
                  style={{
                    display: 'grid',
                    marginRight: '20px'
                  }}
                >
                  {rowLabels.map((label, index) => (
                    <div
                      key={index}
                      style={{
                        alignItems: 'center',
                        display: 'flex',
                        justifyContent: 'center'
                      }}
                      dangerouslySetInnerHTML={{
                        __html: label
                      }}
                    />
                  ))}
                </div>
              )
            }
            <Categories
              model={model}
              disabled={model.disabled}
              categories={categories}
              onDropChoice={this.dropChoice}
              onRemoveChoice={this.removeChoice}
              grid={grid}
            />
          </div>
          <Choices
            disabled={model.disabled}
            model={model}
            choices={choices}
            choicePosition={choicePosition}
          />
        </div>
        {
          model.rationale && (
            <Collapsible
              labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}
              className={classes.collapsible}
            >
              <div dangerouslySetInnerHTML={{ __html: model.rationale }}/>
            </Collapsible>
          )
        }
        {
          model.correctness &&
          model.feedback &&
          !showCorrect && (
            <Feedback
              correctness={model.correctness}
              feedback={model.feedback}
            />
          )
        }
      </div>
    );
  }
}

class CategorizeProvider extends React.Component {
  constructor(props) {
    super(props);
    this.uid = uid.generateId();
  }

  render() {
    return (
      <uid.Provider value={this.uid}>
        <Categorize { ...this.props } />
      </uid.Provider>
    );
  }
}

const styles = (theme) => ({
  prompt: {
    marginBottom: '35px',
    verticalAlign: 'middle'
  },
  categorize: {
    marginBottom: theme.spacing.unit,
    display: 'flex',
    flexDirection: 'column'
  },
  collapsible: {
    paddingBottom: theme.spacing.unit * 2
  }
});
export default withDragContext(withStyles(styles)(CategorizeProvider));
