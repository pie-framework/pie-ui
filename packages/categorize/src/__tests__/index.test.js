import React from 'react';
import Categorize from '../index';
import { shallow } from 'enzyme';
import {
  ModelSetEvent,
  SessionChangedEvent
} from '@pie-framework/pie-player-events';
import { Categorize as UnStyledCategorize } from '../categorize/index';

jest.mock('@pie-lib/math-rendering', () => ({ renderMath: jest.fn() }));

describe('categorize', () => {
  describe('renders', () => {
    let wrapper = props => {
      let defaultProps = {
        model: {
          categories: [],
          choices: [],
          correctResponse: [],
          ...props
        },
        session: {},
        classes: {}
      };
      return shallow(<UnStyledCategorize { ...defaultProps } />)
    };

    it('snapshot', () => {
      const w = wrapper();
      expect(w).toMatchSnapshot();
    });

    it('snapshot with rationale', () => {
      const w = wrapper({ rationale: 'This is rationale'});
      expect(w).toMatchSnapshot();
    });

    it('snapshot with teacherInstructions', () => {
      const w = wrapper({ teacherInstructions: 'These are teacher instructions'});
      expect(w).toMatchSnapshot();
    });
  });

  describe('events', () => {
    describe('model', () => {
      it('dispatches model set event', () => {
        const el = new Categorize();
        el.tagName = 'categorize-el';
        el.model = {};
        expect(el.dispatchEvent).toBeCalledWith(
          new ModelSetEvent('categorize-el', false, true)
        );
      });
    });
    describe('changeAnswers', () => {
      it('dispatches session changed event', () => {
        const el = new Categorize();
        el.tagName = 'categorize-el';
        el.session = { answers: [] };
        el.changeAnswers([{ foo: 'bar' }]);
        expect(el.dispatchEvent).toBeCalledWith(
          new SessionChangedEvent('categorize-el', true)
        );
      });
    });
  });
});
