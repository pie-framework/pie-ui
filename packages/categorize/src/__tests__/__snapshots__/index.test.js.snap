// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categorize renders snapshot 1`] = `
<div>
  <br />
  <_default
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <div
    style={
      Object {
        "flexDirection": "column-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
    </div>
    <WithStyles(Choices)
      choicePosition="above"
      choices={Array []}
      model={
        Object {
          "categories": Array [],
          "choices": Array [],
          "correctResponse": Array [],
        }
      }
    />
  </div>
</div>
`;

exports[`categorize renders snapshot with rationale 1`] = `
<div>
  <br />
  <_default
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <div
    style={
      Object {
        "flexDirection": "column-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
            "rationale": "This is rationale",
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
    </div>
    <WithStyles(Choices)
      choicePosition="above"
      choices={Array []}
      model={
        Object {
          "categories": Array [],
          "choices": Array [],
          "correctResponse": Array [],
          "rationale": "This is rationale",
        }
      }
    />
  </div>
  <Component
    labels={
      Object {
        "hidden": "Show Rationale",
        "visible": "Hide Rationale",
      }
    }
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "This is rationale",
        }
      }
    />
  </Component>
</div>
`;

exports[`categorize renders snapshot with teacherInstructions 1`] = `
<div>
  <Component
    labels={
      Object {
        "hidden": "Show Teacher Instructions",
        "visible": "Hide Teacher Instructions",
      }
    }
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "These are teacher instructions",
        }
      }
    />
  </Component>
  <br />
  <_default
    onToggle={[Function]}
    show={false}
    toggled={false}
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <div
    style={
      Object {
        "flexDirection": "column-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
            "teacherInstructions": "These are teacher instructions",
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
    </div>
    <WithStyles(Choices)
      choicePosition="above"
      choices={Array []}
      model={
        Object {
          "categories": Array [],
          "choices": Array [],
          "correctResponse": Array [],
          "teacherInstructions": "These are teacher instructions",
        }
      }
    />
  </div>
</div>
`;
